### ISSUE TITLE
Error with the example

### ISSUE THREAD

**Issue by Brownie2002** [joshuamkite/terraform-aws-ssh-bastion-service#55]
Hello,

I try to use your module. But for now, I am not able. I tried the example (with adding my credentials) and I get the same error.

```
Error: Error creating Auto Scaling Group: InvalidQueryParameter: Incompatible launch template: Each network interface requires a unique device index.
 status code: 400, request id: ea9c7cfb-fe66-4896-9e67-cd35d45e20d2

with module.ssh-bastion-service.aws_autoscaling_group.bastion-service,
on .terraform\modules\ssh-bastion-service\main.tf line 74, in resource "aws_autoscaling_group" "bastion-service":
74: resource "aws_autoscaling_group" "bastion-service" {
```

I am very noob to terraform, so I try to reproduce the example without success. So maybe I done something wrong.

Regards,

Brownie

**Comment by joshuamkite** [comment]
Hi

Sorry to hear of this- are you able to provide the code you are using to call this module?

Joshua

**Comment by Brownie2002** [comment]
Hello, easy, it is the example : https://github.com/joshuamkite/terraform-aws-ssh-bastion-service/tree/master/examples/full-with-public-ip

So no modification ...

**Comment by joshuamkite** [comment]
I am afraid that I don't see this error when running the example as provided with no changes- deploys without issue for me. Are you adding any optional values such as additional security groups?

**Comment by Brownie2002** [comment]
Hello, unfortunately no ... I will try again later. So, we could close the issue since you cannot reproduce it :(

**Comment by joshuamkite** [comment]
I don't know what to say- I don't really want to dismiss your difficulty as 'works on my machine' but without additional detail or being able to reproduce it I don't know how I could proceed

**Comment by karmajunkie** [comment]
I'm also seeing this issue in my own use. It appears to be related to https://github.com/hashicorp/terraform-provider-aws/issues/23766.  I've installed v4.5 of the `hashicorp/aws` provider (per the linked issue) but to fix the issue you have to tear down the launch template created by v4.6 and let it recreate.

**Comment by joshuamkite** [comment]
Thanks for this @karmajunkie ! IIRC this is a manifestation of an upstream bug with version 4.60 specifically of the AWS provider. Although it is fixed with 4.70 and above these will still break if you have a pre-existing launch template created with 4.6? 

Assuming above is correct, the best I could do here is advise a workaround procedure for 4.6 **but** this would only be relevant for people using or who had used the [4.6 provider, which was only 'current' from March 18, 2022 to March 24, 2022](https://github.com/hashicorp/terraform-provider-aws/releases), i.e. 5 days. Whilst I am happy to be challenged on this, I don't think that this is appropriate or worthwhile to do so within this project and propose to close as 'upstream bug now fixed'

**Comment by joshuamkite** [comment]
Closing as 'upstream bug wontfix (now fixed upstream)'
